csaf-rs/src/csaf/validation.rs

```diff
@@ -19,7 +20,36 @@ impl std::fmt::Display for ValidationError {
     }
 }
 
-#[derive(Debug, PartialEq, Eq, Hash, Clone)]
+/// Result of executing a single test
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct TestResult {
+    /// The test ID that was executed
+    pub test_id: String,
+    /// Whether the test passed
+    pub success: bool,
+    /// The errors if the test failed (empty if successful)
+    pub errors: Vec<ValidationError>,
+}
+
+/// Result of a CSAF validation
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ValidationResult {
+    /// Whether the validation was successful (no errors)
+    pub success: bool,
+    /// The detected CSAF version
+    pub version: String,
+    /// List of validation errors (empty if successful)
+    pub errors: Vec<ValidationError>,
```

Comment:
I don't see any reading usage of this field, and it is just cloned from all the errors within `test_results`, so I think we can remove it.

Reply:
I replaced it with a `num_errors` field, which I still think is convinient to have.